crates/containerd-shim-wamr/src/instance.rs

```diff
@@ -0,0 +1,97 @@
+use anyhow::{Context, Result};
+use containerd_shim_wasm::container::{Engine, Entrypoint, Instance, RuntimeContext, Stdio};
+use wamr_rust_sdk::function::Function;
+use wamr_rust_sdk::instance::Instance as WamrInstnace;
+use wamr_rust_sdk::module::Module;
+use wamr_rust_sdk::runtime::Runtime;
+use wamr_rust_sdk::wasi_context::WasiCtxBuilder;
+
+pub type WamrInstance = Instance<WamrEngine>;
+
+pub struct WamrEngine {
+    runtime: Runtime,
+}
+
+unsafe impl Send for WamrEngine {}
+unsafe impl Sync for WamrEngine {}
+
+// TODO: wasmr_rust_sdk::runtime::Runtime should implement Clone
+
+impl Default for WamrEngine {
+    fn default() -> Self {
+        let runtime = Runtime::new().unwrap();
+        Self { runtime }
+    }
+}
+
+impl Clone for WamrEngine {
+    fn clone(&self) -> Self {
+        let runtime = Runtime::new().unwrap();
+        Self { runtime }
+    }
+}
```

Comment:
We want the `Engine` to be `Clone` because we want cheap copies of the Runtime for each container in the pod. This is the case for Wasmtime, WasmEdge and Wasmer. I believe it should be the case for Wamr as well.

Reply:
yeah I agree with you that we clone the entire process so there is no point to clone the engine. 

> In any case, if we create a new runtime with each clone,

at least for wasmtime, `Engine::clone()` is a cheap operation: https://docs.wasmtime.dev/api/wasmtime/struct.Engine.html#engines-and-clone